### PR TITLE
Store Multiple Lgrs

### PR DESCRIPTION
**Pull request by goncalvesmachadoc**
Create a class to collect and store all LGR objects.

### CODE REVIEW

**Comment by goncalvesmachadoc** [comment]
jenkins build this please

**Comment by goncalvesmachadoc** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/LgrCollection.hpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    void addLgr(const EclipseGrid& grid, const DeckRecord& lgrRecord);
+
+    template<class Serializer>
+    void serializeOp(Serializer& serializer)
+    {
+        serializer(m_lgrs);
+    }
+
+private:
+    OrderedMap<Carfin, 8> m_lgrs;
```
Comment (bska) [inline opm/input/eclipse/EclipseState/Grid/LgrCollection.hpp:56]: I think this is a good idea, but the [`CarFin`](https://github.com/OPM/opm-common/blob/ee44c0fbe5c4c6e9fd86c0fca8de7c4ffc63859c/opm/input/eclipse/EclipseState/Grid/Carfin.hpp#L33) type does not meet every requirement of `OrderedMap` in parallel. In particular the element type `T`&mdash;`CarFin` in this instance&mdash;must have a default constructor for the parallel serialisation code to work.  You will also need to add a `serializeOp()` member function to `CarFin`, and an equality operator (`bool operator==(const LgrCollection&) const`) to the `LgrCollection`.

There should probably be a `serializationTestObject()` function too, but that's not an immediate requirement.

File: src/opm/input/eclipse/EclipseState/Grid/LgrCollection.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    /*
+    This class stores all LGR's. At this moment we only support lgr's input from CARFIN blocks
+    TODO: Collect also lgrs from RADFIN blocks...
+     */
+
+    LgrCollection::LgrCollection(const GRIDSection& gridSection, const EclipseGrid& grid) {
+        const auto& lgrKeywords = gridSection.getKeywordList<ParserKeywords::CARFIN>();
+
+        for (auto keyword_iter = lgrKeywords.begin(); keyword_iter != lgrKeywords.end(); ++keyword_iter) {
+            const auto& lgrsKeyword = *keyword_iter;
```
Comment (bska) [inline src/opm/input/eclipse/EclipseState/Grid/LgrCollection.cpp:43]: Could you write this as
```C++
for (const auto& lgrsKeyword : lgrKeywords) {
```
instead?

**Review by bska — COMMENTED** [review]
This is a good start and I like the direction it's going.  I notice a couple of missing features that are needed in parallel and a couple of small refinements that will make the initialisation easier to read.

**Comment by goncalvesmachadoc** [comment]
jenkins build this please

**Comment by goncalvesmachadoc** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/LgrCollection.hpp (hunk 1 above)
Comment (goncalvesmachadoc) [reply opm/input/eclipse/EclipseState/Grid/LgrCollection.hpp:60]: Thank you for the guidance @bska. I can now compile opm-simulators and pass jenkins tests. Anyway, I am not confident I have added everything needed regarding parallel serialization...will wait for your comments. :)

File: src/opm/input/eclipse/EclipseState/Grid/Carfin.cpp [hunk 3]
```diff
@@ -75,6 +75,15 @@ namespace {
 
 namespace Opm
 {
+
+    Carfin::Carfin(const std::string& lgrName) :
+        m_dims({2,2,2}),   
+        m_offset({1,1,1}),   
+        m_end_offset({2,2,2}),
```
Comment (bska) [inline src/opm/input/eclipse/EclipseState/Grid/Carfin.cpp:82]: How did you select these values?  Are they reasonable defaults for all `Carfin` objects?

**Review by bska — COMMENTED** [review]
I think is good for the most part.  The only piece that I don't quite understand is the single-argument `Carfin::Carfin(const std::string&)` constructor.  Do we need that one?  If not, I think I'd prefer if it were omitted.

**Comment by goncalvesmachadoc** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**